contents/docs/migrate/migrate-between-posthog-instances.mdx

```diff
@@ -42,4 +41,14 @@ Follow the instructions on the [PostHog migrate metadata repo](https://github.co
     ts-node --source [posthog instance you want to migrate from] --sourcekey [personal api key for that instance] --destination [posthog instance you want to migrate to.] --destinationkey [personal api key for destination instance]
     ```
 
-For more information on the options see the [repo's readme](https://github.com/PostHog/posthog-migrate-meta)
\ No newline at end of file
+For more information on the options see the [repo's readme](https://github.com/PostHog/posthog-migrate-meta)
+
+## Migrating your custom apps if you are moving from self-hosted to cloud
+
+Have a custom PostHog app on your self-hosted instance that you'd like to move to the cloud? There are several options:
+1. (Recommended) If you can make your app generalizable enough that others can benefit then [submit your app](https://posthog.com/docs/apps/build/tutorial#submitting-your-app) to the store.
+   - To make it generalizable you'll want to convert anything which is specific to your configuration into a plugin.json config value
+2. If you can't make your app generalizable, you have several options:
+   1. If the app was transforming events before ingestion you could move this logic from the app into your client before you send the event
```

Comment:
There's a third case of apps: import apps like shopify and hubspot. These should be cronjob on their infrastructure, that send events to capture.